src/lib.rs

```diff
@@ -340,6 +333,12 @@ impl Bump {
                 new_size_without_footer = new_size_without_footer.max(requested_size);
             }
 
+            // If we are in the range of using entire pages, let's actually use entire pages
+            if new_size_without_footer >= PAGE_STRATEGY_CUTOFF {
+                new_size_without_footer =
```

Comment:
This will work well if the allocator is storing its metadata about allocations in a side table, but if it is storing them in a header or footer, we are going to get something akin to [clown shoes in sizes 2^10+1 and up](https://blog.mozilla.org/nnethercote/2011/08/05/clownshoes-available-in-sizes-2101-and-up/). That is, the allocator needs a little space to keep track of these allocations, so if the allocation is perfectly multiples-of-page-sized, we can end up allocating an extra page that is almost all unused.

If we are going to go to the trouble of rounding to multiples of a page, lets also remove `MALLOC_OVERHEAD` from the total size (if we can without going under the requested allocation size), which should alleviate this problem.

Reply:
Please take another look. I think I also managed to make the code simpler.

According to the page you linked, we should potentially start rounded to a megabyte bounded after we hit the first megabyte as well.

Reply:
During normal use, we will hit 1m alignment by just doubling, but if a user does either `with_capacity` or allocates something very large, we might only be 4k aligned instead of 1m aligned.

Reply:
I don't think we should over-fit to jemalloc's size classes (especially not the exact size classes reported in that post, since that is (a) a few years old, and (b) specific to mozjemalloc, a fork of jemalloc from even a few years earlier than that)

I care quite a bit about working well with dlmalloc (the default on most "interesting", non-mainstream targets such as wasm32-unknown-unknown) as well as wee_alloc.

I think the page rounding should be a happy place for all of them.

Reply:
Sounds good.